47172 close attachment streams

ricellis opened the pull request:
*What*
Close attachment streams after calculating sha1 hash

*Why*
If the file streams are not closed it causes AttachmentNotSavedExceptions on Windows when deletion of the temporary copy is attempted.

*How*
Add finally blocks to close the streams.

*Testing*
Exercised by existing attachment tests.

reviewer @rhyshort 
reviewer @tomblench

tomblench commented on sync-core/src/main/java/com/cloudant/sync/datastore/PreparedAttachment.java:
```diff
@@ -1,11 +1,11 @@
 /**
  * Copyright (c) 2014 Cloudant, Inc. All rights reserved.
- *
+ * <p/>
```
Do we need these para breaks? If we are concerned about how these comments like in javadoc they should probably all be wrapped in a <pre> block or something. The copyright headers probably all need revisiting soon to fix IBM/Cloudant attributions and other small variations.

tomblench commented on sync-core/src/test/java/com/cloudant/sync/datastore/AttachmentTest.java:
```diff
@@ -42,27 +43,34 @@ public void setAndGetAttachmentsTest() throws Exception {
-            Assert.fail("FileNotFoundException thrown "+fnfe);
+            Assert.fail("FileNotFoundException thrown " + fnfe);
+        } finally {
+            if (fis != null) {
```
Can we use `closeQuietly` here like you did before, and then you don't need to do the null check?

tomblench commented:
Apart from my comments I'm +1 on this, it needs to be done even though java makes the code so much uglier. There's a lot of whitespace changes in this commit which makes it difficult to see what is going on - if you really wanted to apply those changes perhaps they should be in a separate commit?

rhyshort commented:
+1 once tom's comments are resolved

ricellis commented:
In 4b6080d I've fixed up the formatting so that it only applied to the changes instead of also formatting the javadoc.
In 4d56f78 I've switched the test class to also use IOUtils.closeQuietly
